examples/resnet/cifar_input.py

```diff
@@ -30,25 +29,24 @@ def build_data(data_path, size, dataset):
     depth = 3
     image_bytes = image_size * image_size * depth
     record_bytes = label_bytes + label_offset + image_bytes
-
-    data_files = tf.gfile.Glob(data_path)
-    file_queue = tf.train.string_input_producer(data_files, shuffle=True)
+    def load_transform(value):
+        # Convert these examples to dense labels and processed images.
+        record = tf.reshape(tf.decode_raw(value, tf.uint8), [record_bytes])
+        label = tf.cast(tf.slice(record, [label_offset], [label_bytes]), tf.int32)
+        # Convert from string to [depth * height * width] to
+        # [depth, height, width].
+        depth_major = tf.reshape(tf.slice(record, [label_bytes], [image_bytes]),
+                     [depth, image_size, image_size])
+        # Convert from [depth, height, width] to [height, width, depth].
+        image = tf.cast(tf.transpose(depth_major, [1, 2, 0]), tf.float32)
+        return (image, label)
     # Read examples from files in the filename queue.
-    reader = tf.FixedLengthRecordReader(record_bytes=record_bytes)
-    _, value = reader.read(file_queue)
-
-    # Convert these examples to dense labels and processed images.
-    record = tf.reshape(tf.decode_raw(value, tf.uint8), [record_bytes])
-    label = tf.cast(tf.slice(record, [label_offset], [label_bytes]), tf.int32)
-    # Convert from string to [depth * height * width] to
-    # [depth, height, width].
-    depth_major = tf.reshape(tf.slice(record, [label_bytes], [image_bytes]),
-                             [depth, image_size, image_size])
-    # Convert from [depth, height, width] to [height, width, depth].
-    image = tf.cast(tf.transpose(depth_major, [1, 2, 0]), tf.float32)
-    queue = tf.train.shuffle_batch([image, label], size, size, 0,
-                                   num_threads=16)
-    return queue
+    data_files = tf.gfile.Glob(data_path)
+    data = tf.contrib.data.FixedLengthRecordDataset(data_files, record_bytes=record_bytes)
```

Comment:
Out of curiosity, what version of TF does this require?

Reply:
Version 1.2 appears to be the first version with Datasets. I'll update the version check to require at least v1.2.